feat(docs): add webhook documentation (#267)
smartlook/smartlook.github.io: src/webhooks/overview.mdx

---
name: Overview
menu: WEBHOOKS
route: /docs/webhooks/overview
---

import { Title, Code, TextBlock } from "components"
import webhook_settings from 'assets/webhook_settings.png'
import webhooks_settings_anomaly_1 from 'assets/webhooks_settings_anomaly_1.png'

<Title>Overview</Title>
Webhooks are essential part of our API. In opposite to the REST API, that returns data upon a request, webhooks deliver data to you proactively whenever something happens in your project. Currently the webhooks can be triggered by <a href="https://updates.smartlook.com/worried-about-anomalies-in-your-events-190172">anomalies</a> in your data.

# Cookbook

### 1. Set webhook URL on anomalies

> <p>Webhooks you can set in the <a href="https://app.smartlook.com" target="_blank">APP</a>. Below is screen how to set webhooks.</p>

Reviewer: Je dobry se vyvarovat zkratkam a napsat ty veci tak jak jsou. A taky neni potreba CAPSLOCK - prvky se rozlisi pomoci spravnych CSS.

```suggestion
<p>Webhooks you can set in the <a href="https://app.smartlook.com" target="_blank">Smartlook application</a>. Below is screen how to set webhooks.</p>
```

---
name: Overview
menu: WEBHOOKS
route: /docs/webhooks/overview
---

import { Title, Code, TextBlock } from "components"
import webhook_settings from 'assets/webhook_settings.png'

<Title>Overview</Title>

> Webhooks are sent from anomalies. Anomalies are described <a href="https://updates.smartlook.com/worried-about-anomalies-in-your-events-190172" target="_blank">HERE</a>

Reviewer: Tady bych to napsal spis s nejakym lehcim intrem do toho, co vlastne webhooks jsou:

Webhooks are essential part of our API. In opposite to the REST API, that returns data upon a request, webhooks deliver data to you proactively whenever something happens in your project. Currently the webhooks can be triggered by [anomalies](https://updates.smartlook.com/worried-about-anomalies-in-your-events-190172) in your data.

# Cookbook

<h3> 1. Set webhook URL on anomalies </h3>

> Optionaly set secret

Reviewer: Nesel by tady pridat link a mozna jeste popis, kde to vlatne v dashboardu najdou?

Author: dal jsem link a screen :)

<img visibleOn="android,ios" src={webhook_settings} />

<h3> 2. Set API on the server side </h3>

> Webhooks are sent as POST HTTP request. In the section [parameters](#parameters) is more information about webhooks message.

Reviewer: ```suggestion
Webhooks are sent as a JSON in a POST HTTP request to the endpoint you specified in the first step. In the section [parameters](#parameters) is more information about webhooks message.
```

Plus bych tu sekci parameters mozna vynechal a dal ten payload primo sem do textu.

Author: jojo, dava smysl, ze parametry vyhodim hned k tomuto. Zmeneno

<h3> 3. Verify the message </h3>

You can verify the message about HASH. More info is below.

> # Security

Reviewer: Tady bych to klidne vynechal a nechal tady jen ten popis "When your secret token...".

> When your secret token is set, Smartlook uses it to create a hash signature with each payload, in the header.

Reviewer: ```suggestion
When your secret token is set, Smartlook uses it to create a hash signature that we attach to each payload in the header `<DOPLNIT HEADER NAME>`.
```

> <h3> HASH </h3>

Reviewer: ```suggestion
### Hash verification
```

Author: Zmeneno vsude...

> Is used HMAC hex digest to compute the hash.
> You can configure the hash about the code below.

Reviewer: ```suggestion
The Hash-based message authentication codes (HMAC) uses SHA256 hashing function and is digested as a HEX string. Internally the function looks like following:
```

```jsx title="/src/components/hmac.js"
const crypto = require('crypto')

const secret = SECRET_TOKEN
const message = request.body

const signature = crypto
		.createHmac('sha256', secret)
		.update(`${message}`)
		.digest('hex')
```

# Parameters

Example of webhook body
```
{
    "template": "Absolute threshold",
    "anomalyEventName": "Errors",
    "params": {
        "projectName": "Smartlook Project",

>         "projectId": "8588faeebe0e42150310a97a",

Reviewer 2: sou obe ty IDcka vymyslene jo? @Blackman37

Author: vzal jsem puvodni ID a zamenil jsem znaky - pro jistotu jsem to jeste tedka predelal jednou...
